controllers/etcdpeer_controller.go

```diff
@@ -16,14 +20,104 @@ type EtcdPeerReconciler struct {
 	Log logr.Logger
 }
 
+func ignoreNotFound(err error) error {
+	if apierrs.IsNotFound(err) {
+		return nil
+	}
+	return err
+}
+
+const (
+	EtcdImage = "quay.io/coreos/etcd:v3.2.27"
+)
+
 // +kubebuilder:rbac:groups=etcd.improbable.io,resources=etcdpeers,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=etcd.improbable.io,resources=etcdpeers/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=apps,resources=replicaset,verbs=get;update;patch;create
+
+func defineReplicaSet(etcdPeer etcdv1alpha1.EtcdPeer) (appsv1.ReplicaSet, error) {
+	var replicas int32 = 1
+
+	return appsv1.ReplicaSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels:      make(map[string]string),
+			Annotations: make(map[string]string),
+			Name:        etcdPeer.Name,
+			Namespace:   etcdPeer.Namespace,
+		},
```

Comment:
You should also add an OwnerReference to this ReplicaSet, and then call `Owns` when constructing the controller: https://github.com/munnerz/etcd-operator-poc/blob/6f8e92404b3bc89bdef60bea195e9fb111ba12ad/controllers/etcdpeer_controller.go#L249-L250

```
	return ctrl.NewControllerManagedBy(mgr).
		For(&etcdv1alpha1.EtcdPeer{}).
		// Watch for changes to ReplicaSet resources that an EtcdPeer owns.
		Owns(&core.ReplicaSet{})
```